File: javascript/rna-transcription/rna-transcription.ts

class DnaTranscriber {
    constructor(){}
    toRna(sequence: string): string {
      let sequenceArray: Array<string> = sequence.split("");
      return sequenceArray.map(this.getComplement).join("");
    }

    getComplement(nucleotide: string): string {
      let complementPairs = {
        'G': 'C',
        'C': 'G',
        'T': 'A',
        'A': 'U'
      };
      return complementPairs[nucleotide];
    }
}


module.exports = DnaTranscriber;

File: python/rna-transcription/rna_transcription.py
def to_rna(dna_strand):
    rna_chars = [dna_to_rna[char] for char in dna_strand]
    return "".join(rna_chars)

dna_to_rna = {
    "G": "C",
    "C": "G",
    "T": "A",
    "A": "U",
}
